tests/test_precision.py

```diff
@@ -55,3 +67,133 @@ def test_distace_profile():
         )
 
         npt.assert_almost_equal(D_ref, D_comp)
+
+
+def test_calculate_squared_distance():
+    # This test function raises an error if the distance between a subsequence
+    # and another does not satisfy the symmetry property.
+    seed = 332
+    np.random.seed(seed)
+    T = np.random.uniform(-1000.0, 1000.0, [64])
+    m = 3
+
+    T_subseq_isconstant = core.rolling_isconstant(T, m)
+    M_T, Σ_T = core.compute_mean_std(T, m)
+
+    n = len(T)
+    k = n - m + 1
+    for i in range(k):
+        for j in range(k):
+            QT_i = core._sliding_dot_product(T[i : i + m], T)
+            dist_ij = core._calculate_squared_distance(
+                m,
+                QT_i[j],
+                M_T[i],
+                Σ_T[i],
+                M_T[j],
+                Σ_T[j],
+                T_subseq_isconstant[i],
+                T_subseq_isconstant[j],
+            )
+
+            QT_j = core._sliding_dot_product(T[j : j + m], T)
+            dist_ji = core._calculate_squared_distance(
+                m,
+                QT_j[i],
+                M_T[j],
+                Σ_T[j],
+                M_T[i],
+                Σ_T[i],
+                T_subseq_isconstant[j],
+                T_subseq_isconstant[i],
+            )
+
+            comp = dist_ij - dist_ji
+            ref = 0.0
+
+            assert ref == comp
+            # npt.assert_almost_equal(ref, comp, decimal=15)
```

Comment:
Please use the `decimal=15` version

Reply:
Otherwise, the unit test is still failing:

https://github.com/TDAmeritrade/stumpy/actions/runs/5773239904/job/15648976680?pr=892

Reply:
Doesn't unit test may still fail since difference is in the order of `1e-15`? It fails in **minimum** version testing.

Reply:
I updated the branch and applied the changes, and pushed the commit. Going to have my eye on the tests to see how they go.

Reply:
[Before I forget]
As one of our last resorts, we may remove the flag `reassoc` from fastmath and see if it solves the issue and [if yes] how much it affects the running time.

Reply:
Sounds good!